### PR TITLE
Updating Docker files for UPS

### PR DESCRIPTION
**Pull request by matzew**
* removing quickstarts image
* Using Centos7, WF9 and Java8 (inheriting from 9.0.0.Final WF image) for UPS images 
* explicitly referencing 9.0.0.Final instead of latests
* DB references via container links for MySQL
* README updates

### CODE REVIEW

**Comment by matzew** [comment]
@danielpassos @secondsun @sebastienblanc @abstractj wanna take a look ?

**Comment by secondsun** [comment]
followed the instructions in the readme, got this when I fired up the UPS container
https://gist.github.com/secondsun/e1a7d791636c5b448994

**Comment by matzew** [comment]
did you do the linking ?

On Wed, Jul 22, 2015 at 5:06 PM, Hoyt Summers Pittman <
notifications@github.com> wrote:

> followed the instructions in the readme, got this when I fired up the UPS
> container
> https://gist.github.com/secondsun/e1a7d791636c5b448994
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/dockerfiles/pull/8#issuecomment-123752853>.
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf

**Comment by danielpassos** [comment]
My 2 cents https://github.com/matzew/dockerfiles/pull/1

**Comment by secondsun** [comment]
@matzew 
Looking higher up in the thread looks like i had this : https://gist.github.com/secondsun/e2b25d0e22c62a13971d

**Comment by matzew** [comment]
summers,

you need to build the wildfly image too!
not just the dev!

your is still java7

mine goes Centos7, wf9 and jdk8 ;)

On Wednesday, July 22, 2015, Hoyt Summers Pittman <notifications@github.com>
wrote:

> @matzew <https://github.com/matzew>
> Looking higher up in the thread looks like i had this :
> https://gist.github.com/secondsun/e2b25d0e22c62a13971d
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/dockerfiles/pull/8#issuecomment-123805451>.
>


-- 
Sent from Gmail Mobile

File: wildfly/unifiedpush-wildfly-dev/README.md [hunk 1]
```diff
@@ -8,9 +8,43 @@ Follow the [instructions](http://docs.docker.com/installation/)
...
+           -e MYSQL_DATABASE=unifiedpush \
+           -e MYSQL_ROOT_PASSWORD=supersecret \
+           -d mysql
+```           
+
+The two databases are now linked into the container that serves WildFly, containing the latest release of the UPS
+
+```shell
+docker run --name ups-dev \
+           --link unifiedpushDEV:unifiedpush
```
Comment (secondsun) [inline wildfly/unifiedpush-wildfly-dev/README.md]: missing a '\' after unifiedpush

Comment (danielpassos) [inline wildfly/unifiedpush-wildfly-dev/README.md]: Good catch

**Comment by danielpassos** [comment]
```
19:14:54,477 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-6) MSC000001: Failed to start service jboss.deployment.unit."ag-push.war".PARSE: org.jboss.msc.service.StartException in service jboss.deployment.unit."ag-push.war".PARSE: JBAS018733: Failed to process phase PARSE of deployment "ag-push.war"
	at org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:166) [wildfly-server-8.2.0.Final.jar:8.2.0.Final]
```

What I did?

```shell
docker build -t aerogear/unifiedpush-wildfly .

docker run --name keycloak \
           -p 5306:3306 \
           -e MYSQL_USER=unifiedpush \
           -e MYSQL_PASSWORD=unifiedpush \
           -e MYSQL_DATABASE=unifiedpush \
           -e MYSQL_ROOT_PASSWORD=supersecret \
           -d mysql        

docker run --name unifiedpush \
           -p 3306:3306 \
           -e MYSQL_USER=unifiedpush \
           -e MYSQL_PASSWORD=unifiedpush \
           -e MYSQL_DATABASE=unifiedpush \
           -e MYSQL_ROOT_PASSWORD=supersecret \
           -d mysql

docker run --name ups \
           --link unifiedpush:unifiedpush \
           --link keycloak:keycloak \
           -p 8443:8443 \
           -it aerogear/unifiedpush-wildfly
```

**Comment by danielpassos** [comment]
> you need to build the wildfly image too! not just the dev!

So, maybe is a good idea add it in the README

**Comment by danielpassos** [comment]
After build `aerogear/wildfly` first. It works for me

**Comment by vibe13** [comment]
@matzew Running the second run command (first went fine), I get:

FATA[0002] Error response from daemon: Cannot start container 0916d9f7e2498c42dfcdd9404aa439ce101674a4f1a3bcd84a4536947d2b581a: Error starting userland proxy: listen tcp 0.0.0.0:3306: bind: address already in use 

[ Fedora 21, Docker version 1.6.0, build 350a636/1.6.0]

**Comment by vibe13** [comment]
@matzew 
Sorry, I built the first image (wildfly) without specifying the tag. I also had a local MySQL instance running, which I stopped.
Rebuilding the 2 images with the proper commands, and executing the 3 run commands above, everything worked.
Thanks!

**Comment by vibe13** [comment]
Worth mentioning that there is an error in the log (it does not block the deployment):

10:47:36,114 ERROR [org.hibernate.tool.hbm2ddl.SchemaUpdate] (ServerService Thread Pool -- 67) HHH000388: Unsuccessful: create index deviceTokenIndex on Installation (deviceToken)
10:47:36,114 ERROR [org.hibernate.tool.hbm2ddl.SchemaUpdate] (ServerService Thread Pool -- 67) Specified key was too long; max key length is 767 bytes

**Comment by matzew** [comment]
@vibe13 the INDEX is a known thing - we have JIRA somewhere for that :)

**Comment by vibe13** [comment]
@matzew ok great :-)

**Comment by matzew** [comment]
@vibe13 what Docker version are you using ? @secondsun had issues w/ 1.6.x on his Fedora-21

**Comment by secondsun** [comment]
@matzew I'm running 1.7.1 now.  Still have the same issues.

**Comment by matzew** [comment]
I tested the unifiedpush-wildfly from a VM, that comes w/ Fedora:

```
cat /etc/redhat-release && docker --version
Fedora release 21 (Twenty One)
Docker version 1.5.0, build a8a31ef/1.5.0
```

and, well - works there

**Comment by secondsun** [comment]
Ok so here are my tests :

Fedora 22 Docker provided 1.7.0, 1.7.1 - aerogear/unifiedpush-wildfly can not connect to the mysql instances.
Fedora 22 Fedora provided 1.7.0, 1.7.1  - aerogear/unifiedpush-wildfly can not connect to the mysql instances.

Fedora 21 - Fedora Provided 1.6.0 - runs correctly
Debian Stable - Debian provided 1.5.0 - runs correctly.

**Comment by secondsun** [comment]
One thing that I noticed : If you build from the master branch and someone updates the master branch docker won't pull the updates because it will still be using the cached steps from a previous build.

It looks like there are workarounds to make sure the clone step runs every time, but that will be something to investigate

**Comment by matzew** [comment]
@secondsun mind giving this a new try?

* building the WF image
* building the -dev image 

That step was failing for you in the past

**Comment by secondsun** [comment]
Everything is working for me on Fedora 22 now.

GG

**Comment by secondsun** [comment]
Stable looks good, +1

**Pull request merged**